### PR TITLE
[KERNEL32] Fix argument information for stubs

### PR DESCRIPTION
**Pull request by Getequ**
App can fail when called our stub w/o args but function actually have them and app is pushing data to stack

### CODE REVIEW

**Comment by learn-more** [comment]
What?

**Comment by Getequ** [comment]
@learn-more firefox 48 is calling GetProcessDEPPolicy and failed with 8000100 exception code. Similar result was with WerSetFlags while installing .net 4.5

I built ros with guOsVersion=0x600 in spec2def (and no limit by endversion)

**Comment by encodedpr** [comment]
i like it

File: dll/win32/kernel32/kernel32.spec [hunk 1]
```diff
@@ -162,10 +162,10 @@
 @ stdcall -stub -version=0x600+ CreateSemaphoreExW(ptr long long wstr long long)
 @ stdcall CreateSemaphoreW(ptr long long wstr)
 @ stdcall -i386 CreateSocketHandle()
-@ stdcall -version=0x600+ CreateSymbolicLinkA(str str long)
+@ stdcall -stub -version=0x600+ CreateSymbolicLinkA(str str long)
```
[the remark this answers is not in the document]
Comment (HBelusca) [reply dll/win32/kernel32/kernel32.spec]: What about making a redirection to kernel32_vista then?

[the remark this answers is not in the document]
Comment (tkreuzer) [reply dll/win32/kernel32/kernel32.spec]: It depends on how we want to use this. Currently it is not used at all IIRC. Our versioning system is WIP and should eventually result in one core library with all the exports and a number of wrappers, not different full implementations or a mixture of half implementations and forwarders.

[the remark this answers is not in the document]
Comment (Getequ) [reply dll/win32/kernel32/kernel32.spec]: Oops, sorry. I don't remember how/why I did it :) fixed

**Comment by Getequ** [comment]
@tkreuzer  hi, is all ok atm?)  Yes, almost 2 years passed from PR was opened but i rebased that commit onto master

File: dll/win32/kernel32/kernel32.spec [hunk 2]
```diff
@@ -1148,12 +1148,12 @@
...
 @ stdcall -ret64 VerSetConditionMask(long long long long) ntdll.VerSetConditionMask
 @ stdcall VerifyConsoleIoHandle(long)
-@ stub -version=0x600+ VerifyScripts
+@ stdcall -stub -version=0x600+ VerifyScripts(long wstr long wstr long)
 @ stdcall VerifyVersionInfoA(long long double)
 @ stdcall VerifyVersionInfoW(long long double)
 @ stdcall VirtualAlloc(ptr long long long)
 @ stdcall VirtualAllocEx(long ptr long long long)
-@ stub -version=0x600+ VirtualAllocExNuma
+@ stdcall -stub -version=0x600+ VirtualAllocExNuma(ptr ptr long long long long)
```
Comment (HBelusca) [inline dll/win32/kernel32/kernel32.spec:1156]: 3rd parameter is SIZE_T (and is pointer-sized); may require "ptr" instead

File: dll/win32/kernel32/kernel32.spec [hunk 3]
```diff
@@ -1193,10 +1193,10 @@
 @ stdcall WinExec(str long)
 @ stdcall Wow64DisableWow64FsRedirection(ptr)
 @ stdcall Wow64EnableWow64FsRedirection(long)
-@ stub -version=0x600+ Wow64GetThreadContext
+@ stdcall -stub -version=0x600+ Wow64GetThreadContext(ptr ptr)
```
Comment (HBelusca) [inline dll/win32/kernel32/kernel32.spec:1196]:
```suggestion
@ stdcall -stub -version=0x600+ -arch=x86_64 Wow64GetThreadContext(ptr ptr)
```

File: dll/win32/kernel32/kernel32.spec [hunk 4]
```diff
@@ -1193,10 +1193,10 @@
...
 @ stdcall Wow64RevertWow64FsRedirection(ptr)
-@ stub -version=0x600+ Wow64SetThreadContext
-@ stub -version=0x600+ Wow64SuspendThread
+@ stdcall -stub -version=0x600+ Wow64SetThreadContext(ptr ptr)
```
Comment (HBelusca) [inline dll/win32/kernel32/kernel32.spec:1198]:
```suggestion
@ stdcall -stub -version=0x600+ -arch=x86_64 Wow64SetThreadContext(ptr ptr)
```

File: dll/win32/kernel32/kernel32.spec [hunk 5]
```diff
@@ -1193,10 +1193,10 @@
...
+@ stdcall -stub -version=0x600+ Wow64SuspendThread(ptr)
```
Comment (HBelusca) [inline dll/win32/kernel32/kernel32.spec:1199]:
```suggestion
@ stdcall -stub -version=0x600+ -arch=x86_64 Wow64SuspendThread(ptr)
```

File: dll/win32/kernel32/kernel32.spec [hunk 6]
```diff
@@ -1193,10 +1193,10 @@
 @ stdcall WinExec(str long)
 @ stdcall Wow64DisableWow64FsRedirection(ptr)
```
Comment (HBelusca) [inline dll/win32/kernel32/kernel32.spec:1194]: All those Wow64 functions quite probably need an extra `-arch=x86_64`.
Cc @tkreuzer for confirmation.

**Review by HBelusca — CHANGES_REQUESTED** [review]

**Pull request closed without merging**